### PR TITLE
909 Add NumpyReader for IO factory

### PR DESCRIPTION
**Pull request by Nic-Ma**
Part of #909 

### Description
This PR implemented `NumpyReader` which is part of the IO factory task.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/data/image_reader.py [hunk 1]
```diff
@@ -348,3 +349,94 @@ def _get_array_data(self, img: Nifti1Image) -> np.ndarray:
...
+        if isinstance(data, np.ndarray):
+            self._img.append(data)
+            return data
+
+        filenames: Sequence[str] = ensure_tuple(data)
+        for name in filenames:
+            img = np.load(name, allow_pickle=True, **kwargs)
+            if name.endswith(".npz"):
+                # load expected items from NPZ file
+                npz_keys = [f"arr_{i}" for i in range(len(img))] if self.npz_keys is None else self.npz_keys
```
Comment (ericspod) [inline monai/data/image_reader.py:407]: Is there a reason for restricting keys to "arr_{i}" names  if not given? I would use `npz_keys = list(self.npz_keys or img.keys())`,

Comment (Nic-Ma) [reply monai/data/image_reader.py:407]: Hi @ericspod ,

"arr_{i}" is the default key name of NPZ data, and we want to keep the order and stack the data, so didn't use `img.keys()`. It's the same behavior with our existing `LoadNumpy` transform. Make sense?

Thanks.

Comment (ericspod) [reply monai/data/image_reader.py:407]: The order part of that makes sense. I think `img.keys()` will give you names in order but perhaps best not to rely on it.

**Review by ericspod — APPROVED** [review]

File: monai/data/image_reader.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/data/image_reader.py:407]: Yes, I agree.
Thanks.

**Pull request merged**